### DISCUSSION TITLE
SSR data is fetched again on the client with vike-vue-query

### DISCUSSION THREAD

**Discussion by BMFreed** [vikejs/vike#1792] [Discussion] [answered]
Greetings!
I have noticed a strange behaviour while using vike-vue-query.
I've launched the [example repository](https://github.com/vikejs/vike-vue/tree/main/examples/vue-query) and saw that while the data on the page is prerendered via SSR - it is also fetched after the initial render, while the expected behaviour is that vike-vue-query will tell TanStack Query to hydrate the client instead of refetching data.

The SSR response:
![image](https://github.com/user-attachments/assets/c138ea59-4e4a-4afc-9a29-c47346c3f273)

The repeated request sent from the client side:
![image](https://github.com/user-attachments/assets/39e3734d-fff8-44d7-9f04-be99bba8655e)

Am I getting something wrong?

**Comment by brillout** [comment]
Indeed, the data shouldn't be re-fetched. Can you publish a reproduction? CC @nitedani

**Reply by BMFreed** [reply]
The issue can be reproduced via vike-vue-query's official [example repository](https://github.com/vikejs/vike-vue/tree/main/examples/vue-query)
I've also linked it in the original post. Hope this helps :)

**Comment by BMFreed** [comment, marked as answer]
@brillout I have found the root cause of the issue. Turns out, this is an expected behaviour in TanStack Query, where if you use SSR - you have to set the query's stale time to a high enough value to avoid immediate refetch.
It was actually answered [here](https://github.com/TanStack/query/discussions/4870).
So there is no issue with vike or vike-vue-query whatsoever.

**Reply by brillout** [reply]
@nitedani Maybe `vike-react-query` should add a default stale time?

**Reply by BMFreed** [reply]
> @nitedani Maybe `vike-react-query` should add a default stale time?

Personally, I think this would abstract a part of TanStack Query's API.
However, it would be great if you could add this detail about TanStack Query and SSR to the docs. And also set the stale time in the example repository, while linking the docs in the code :)
